eppo/src/main/java/cloud/eppo/android/dto/deserializers/RandomizationConfigResponseDeserializer.java

```diff
@@ -0,0 +1,195 @@
+package cloud.eppo.android.dto.deserializers;
+
+import static cloud.eppo.android.util.Utils.logTag;
+
+import android.util.Log;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import cloud.eppo.android.EppoClient;
+import cloud.eppo.android.dto.Allocation;
+import cloud.eppo.android.dto.EppoValue;
+import cloud.eppo.android.dto.FlagConfig;
+import cloud.eppo.android.dto.OperatorType;
+import cloud.eppo.android.dto.RandomizationConfigResponse;
+import cloud.eppo.android.dto.ShardRange;
+import cloud.eppo.android.dto.TargetingCondition;
+import cloud.eppo.android.dto.TargetingRule;
+import cloud.eppo.android.dto.Variation;
+import cloud.eppo.android.dto.adapters.EppoValueAdapter;
+
+/**
+ *  Hand-rolled deserializer so that we don't rely on annotations and method names, which can be
+ *  unreliable when ProGuard minification is in-use and not configured to protect
+ *  JSON-deserialization-related classes and annotations.
+ */
+public class RandomizationConfigResponseDeserializer implements JsonDeserializer<RandomizationConfigResponse> {
```

Comment:
I went with `RandomizationConfigResponseDeserializer` instead of `RandomizationConfigResponseAdapter` (the pattern `EppoValueAdapater` followed) as it's implementing the `JsonDeserializer` interface so I felt it was more descriptive, even though its main use is in `gson`'s .`registerTypeAdapter()` method.